File: src/main/java/com/rody/Hello.java
package com.rody;

public class Hello {
    public static void main(String[] args) {
//        System.out.println("Hello world");
//        new Person().hello();
        Person person= new Person();
        person.hello();
        person.hight= 1.7f;
        person.weight= 66.5f;
        System.out.println("person bmi"+person.bim());

        Person person2= new Person(66.5f, 1.7f);
        System.out.println("person2 bmi"+person2.bim());

        Person person3= new Person("Rody",66.5f, 1.7f);
        System.out.println("person3 name:"+person3.name+" bmi"+person3.bim());

        /*
        *   short   16 bits
        *   int     32 bits ==> Integer     ==>Wrapper class
        *   long    64 bits
        *   char    16 bits ==> Character   ==>Wrapper class
        *   byte    8  bits
        *   float   32 bits ==> Float       ==>Wrapper class
        *   double  64 bits ==> Double      ==>Wrapper class
        *   boolean         ==> Boolean     ==>Wrapper class
        *   String  class                    ==> Reference data type
        */
        int age= 19; // <=== zone variable



    }
}

File: src/main/kotlin/com/rody/Hello.kt
package com.rody

fun main(args: Array<String>) {
//    println("Hellow Kotlin")
//    Human().hello()

    val human= Human(66.5f,1.7f)
    //human.hello()
    println(human.bmi())
    // Kotlin 沒有 基本資料型態, 只有物件的相關型態 Ex: Int, Short, Double, Float, Long
    // val --> 定義完物件後不能再改變
    // var \--> 定義完後可以在改變
    // val 名稱:型態=值
    val age : Int =19; //ot val age=19 <==Kotlin 會自己推斷型態
    var name:String
    name="Rody"

    val score=88
    println(score>60)

    val c:Char='A'
    // error println(c>60)
    println(c.toInt()>60)
}

// 主要建構子
class Human(var weight: Float, var height: Float,var name:String="") {

    init {
        println("test $weight")
    }
    // 次要建構子
    //constructor(name:String, weight: Float,height: Float):this(weight,height)

    fun bmi():Float {
        val bmi=weight/(height*height)
        return bmi
    }
    fun hello(){
        println("Hello Kotlin by Human")
    }
}